### PR TITLE
Add Consent Mode section to settings screen.

### PR DESCRIPTION
**Pull request by techanvil**
## Summary

Addresses issue:

- #8278 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/consent-mode/ConfirmDisableConsentModeDialog.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+					listFormat( dependentModuleNames )
+			  )
+			: null;
+
+	return (
+		<ModalDialog
+			dialogActive
+			title={ __( 'Disable consent mode?', 'google-site-kit' ) }
+			subtitle={ __(
+				'Disabling consent mode will remove the tag from your website, you will no longer be able to track these in the European Economic Area:',
```
Comment (tofumatt) [inline assets/js/components/consent-mode/ConfirmDisableConsentModeDialog.js:71]: This one is quite hard to read. "no longer able to track these in" really threw me for a loop.

What id we update this text to:

> Disabling consent mode will remove the tag from your website. In the European Economic Area, you will no longer be able to track:

Comment (tofumatt) [reply assets/js/components/consent-mode/ConfirmDisableConsentModeDialog.js:71]: (We can wait for @marrrmarrr's response for this one and update it in a separate PR if needed, no need to hold up the entire PR, but I think we can improve the clarity of this warning.)

File: assets/js/components/consent-mode/ConsentModeSwitch.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+						<Switch
+							label={ __(
+								'Enable consent mode',
+								'google-site-kit'
+							) }
+							checked={ isConsentModeEnabled }
+							disabled={ loading || isSaving }
+							onClick={ () => {
+								if ( isConsentModeEnabled ) {
+									setShowConfirmDialog( true );
```
Comment (tofumatt) [inline assets/js/components/consent-mode/ConsentModeSwitch.js:87]: It's a bit odd that the "show confirm dialog" is confirming you want to disable Consent Mode (eg. it's a "disable dialog"). 🤷🏻‍♂️

File: assets/js/components/consent-mode/ConsentModeSwitch.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+						height="42px"
+						smallWidth="386px"
+						smallHeight="84px"
+						tabletWidth="540px"
+						tabletHeight="84px"
+					>
+						<p>
+							{ createInterpolateElement(
+								__(
+									'Consent mode will help adjust tracking on your site, so only visitors who have explicitly given consent are tracked.<br />This is required in some parts of the world, like the European Economic Area. <a>Learn more</a>',
```
Comment (tofumatt) [inline assets/js/components/consent-mode/ConsentModeSwitch.js:120]: There should still be a space between words when there's a line break like this. Most software makes sense of it but technically the line break isn't always interpreted as whitespace, so it's best to have both:

```suggestion
									'Consent mode will help adjust tracking on your site, so only visitors who have explicitly given consent are tracked. <br />This is required in some parts of the world, like the European Economic Area. <a>Learn more</a>',
```

File: assets/js/components/settings/SettingsCardConsentMode.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
+/**
+ * Site Kit by Google, Copyright 2023 Google LLC
```
Comment (tofumatt) [inline assets/js/components/settings/SettingsCardConsentMode.js:2]:
```suggestion
 * Site Kit by Google, Copyright 2024 Google LLC
```

😅

**Review by tofumatt — COMMENTED** [review]
Overall looks good, just a few very small points, probably better to handle them as follow-up issues to speed this along 😄

**Review by tofumatt — APPROVED** [review]

File: assets/js/components/consent-mode/ConfirmDisableConsentModeDialog.js (hunk 1 above)
Comment (tofumatt) [reply assets/js/components/consent-mode/ConfirmDisableConsentModeDialog.js:71]: Just chatted with @marrrmarrr on Slack, I'll make these updates here 😄 https://10up.slack.com/archives/CFFRMC5DE/p1709587420597549

Comment (techanvil) [reply assets/js/components/consent-mode/ConfirmDisableConsentModeDialog.js:71]: Awesome, thanks @tofumatt!

File: assets/js/components/consent-mode/ConsentModeSwitch.js (hunk 2 above)
Comment (techanvil) [reply assets/js/components/consent-mode/ConsentModeSwitch.js:87]: That is a fair point, it's a little counter intuitive. It's not tooo bad, and I see you have approved this so in the interests of moving along I think we can consider renaming it in a future iteration.

File: assets/js/components/settings/SettingsCardConsentMode.js (hunk 4 above)
Comment (techanvil) [reply assets/js/components/settings/SettingsCardConsentMode.js]: Those copy/pastas inevitably sneak in :sweat_smile:

File: assets/js/components/consent-mode/ConsentModeSwitch.js (hunk 3 above)
Comment (techanvil) [reply assets/js/components/consent-mode/ConsentModeSwitch.js]: Good shout, cheers.

**Pull request merged**